packages/shared/src/tms/projection.tile.matrix.set.ts

```diff
@@ -0,0 +1,96 @@
+import { Bounds, Epsg, GeoJson, QuadKey, Tile, TileMatrixSet, EpsgCode } from '@basemaps/geo';
+import * as proj4 from 'proj4';
+import { Position } from 'geojson';
+import { Nztm2000 } from './nztm2000';
+import { Citm2000 } from './citm2000';
+import { GoogleTms } from '@basemaps/geo/build/tms/google';
+import { Nztm2000Tms } from '@basemaps/geo/build/tms/nztm2000';
+import { CompositeError } from '../composite.error';
+
+export interface LatLon {
+    lat: number;
+    lon: number;
+}
+
+proj4.defs(Epsg.Nztm2000.toEpsgString(), Nztm2000);
+proj4.defs(Epsg.Citm2000.toEpsgString(), Citm2000);
+
+const CodeMap = new Map<EpsgCode, ProjectionTileMatrixSet>();
+
+export class ProjectionTileMatrixSet {
+    tms: TileMatrixSet;
+    private projection: proj4.Converter;
+
+    constructor(tms: TileMatrixSet) {
+        this.tms = tms;
+        try {
+            this.projection = proj4(tms.projection.toEpsgString(), Epsg.Wgs84.toEpsgString());
+        } catch (err) {
+            throw new CompositeError(
+                `Failed to create projection: ${tms.projection.toEpsgString()}, ${Epsg.Wgs84.toEpsgString()}`,
+                err,
+            );
+        }
+    }
+
+    static get(epsgCode: EpsgCode): ProjectionTileMatrixSet {
+        const ptms = CodeMap.get(epsgCode);
+        if (ptms != null) return ptms;
+        throw new Error(`Invalid projection: ${epsgCode}`);
+    }
+
+    static tryGet(epsgCode?: EpsgCode): ProjectionTileMatrixSet | null {
+        return (epsgCode && CodeMap.get(epsgCode)) ?? null;
+    }
+
+    get forward(): (coordinates: Position) => Position {
```

Comment:
do we want to name these more inline what they are doing, since they are always to/from wgs84

Reply:
eg `sourceToWgs84()` ` wgs84toSource`